coding/wml-v20/src/main/java/org/n52/sos/encode/streaming/WmlTVPEncoderv20XmlStreamWriter.java

```diff
@@ -164,18 +170,18 @@ private void writeMeasurementTimeseriesMetadata(String id) throws XMLStreamExcep
     /**
      * Write wml:defaultPointMetadata to stream
      *
-     * @param unit
+     * @param observation
      * @throws XMLStreamException
      *             If an error occurs when writing to stream
      */
-    private void writeDefaultPointMetadata(String unit) throws XMLStreamException {
+    private void writeDefaultPointMetadata(OmObservation observation) throws XMLStreamException {
         start(WaterMLConstants.QN_DEFAULT_POINT_METADATA);
         writeNewLine();
         start(WaterMLConstants.QN_DEFAULT_TVP_MEASUREMENT_METADATA);
         writeNewLine();
-        writeUOM(unit);
+        writeUOM(observation.getValue().getValue().getUnit());
```

Comment:
As discussed, have a look at why getValue is only called once for StreamingValue in the old code